### PR TITLE
Use upstream swagger

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

**Comment by soltysh** [comment]
You're missing `examples/apiserver/apiserver.go` from the upstream PR.

File: pkg/cmd/server/origin/master.go [hunk 1]
```diff
@@ -332,11 +335,9 @@ func (s sortedGroupVersions) Len() int           { return len(s) }
 func (s sortedGroupVersions) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
 func (s sortedGroupVersions) Less(i, j int) bool { return s[i].Group < s[j].Group }
 
-func (c *MasterConfig) installHandlers(apiContainer *genericmux.APIContainer) []string {
+func (c *MasterConfig) kubernetesExtraMessages() []string {
```
Comment (soltysh) [inline pkg/cmd/server/origin/master.go]: How about `kubernetesAPIMessages`?

File: pkg/cmd/server/origin/master.go [hunk 2]
```diff
@@ -302,12 +303,14 @@ func (c *MasterConfig) RunInProxyMode(proxy *kubernetes.ProxyConfig, assetConfig
...
-	url, err := url.Parse(c.Options.MasterPublicURL)
-	if err != nil {
-		glog.Fatalf("Failed to parse master public url %q: %v", c.Options.MasterPublicURL, err)
-	}
-	genericroutes.Swagger{ExternalAddress: url.Host}.Install(container)
+
+	swaggerConfig := genericapiserver.DefaultSwaggerConfig()
+	swaggerConfig.WebServicesUrl = c.Options.MasterPublicURL
+	genericroutes.Swagger{Config: swaggerConfig}.Install(container)
+	extra = append(extra, fmt.Sprintf("Started Swagger Schema API at %%s%s", swaggerAPIPrefix))
```
Comment (soltysh) [inline pkg/cmd/server/origin/master.go]: You should drop the `swaggerAPIPrefix` and read it from `swaggerConfig.ApiPath`

File: pkg/cmd/server/origin/master.go [hunk 3]
```diff
@@ -132,23 +133,15 @@ import (
 )
 
 const (
-	LegacyOpenShiftAPIPrefix = "/osapi" // TODO: make configurable
-	OpenShiftAPIPrefix       = "/oapi"  // TODO: make configurable
-	OpenShiftAPIV1           = "v1"
-	OpenShiftAPIPrefixV1     = OpenShiftAPIPrefix + "/" + OpenShiftAPIV1
-	swaggerAPIPrefix         = "/swaggerapi/"
-	openAPIServePath         = "/swagger.json"
+	swaggerAPIPrefix = "/swaggerapi/"
```
Comment (soltysh) [inline pkg/cmd/server/origin/master.go]: Get rid of it in favor of reading it from `swaggerConfig.ApiPath`.

File: pkg/cmd/server/origin/master.go [hunk 4]
```diff
@@ -132,23 +133,15 @@ import (
...
+	openAPIServePath = "/swagger.json"
```
Comment (soltysh) [inline pkg/cmd/server/origin/master.go:136]: It's a shame this isn't a constant/config option somewhere upstream like swagger api prefix.

File: pkg/cmd/server/origin/master.go [hunk 5]
```diff
@@ -261,8 +150,6 @@ func (c *MasterConfig) Run(kc *kubernetes.MasterConfig, assetConfig *AssetConfig
 		extra []string
```
Comment (soltysh) [inline pkg/cmd/server/origin/master.go]: That `extra` variable name is like from hell, I'd vote to rename it to something meaningful, like messages, logs, etc.

**Review by soltysh — COMMENTED** [review]
A few nits, but overall this LGTM.

File: pkg/cmd/server/origin/master.go (hunk 2 above)
Comment (sttts) [reply pkg/cmd/server/origin/master.go]: done. The OpenAPI equivalent is hardcoded upstream. So keeping that for now.

File: pkg/cmd/server/origin/master.go (hunk 1 above)
Comment (sttts) [reply pkg/cmd/server/origin/master.go]: `kubernetesAPIExtraMessages`

File: pkg/cmd/server/origin/master.go (hunk 5 above)
Comment (sttts) [reply pkg/cmd/server/origin/master.go]: 120% agree.

**Comment by sttts** [comment]
Addressed all of @soltysh's comments.

**Pull request merged**